libsolidity/formal/CHC.cpp

```diff
@@ -1594,6 +1595,33 @@ void CHC::checkVerificationTargets()
 		checkedErrorIds.insert(target.errorId);
 	}
 
+	auto toReport = m_unsafeTargets;
+	if (m_settings.showUnproved)
+		for (auto const& [node, targets]: m_unprovedTargets)
+			for (auto const& [target, info]: targets)
+				if (!toReport[node].count(target))
+					toReport[node].emplace(target, info);
```

Comment:
This is probably a good place to expand the auto type. Too many types to follow otherwise.

Reply:
Because if there is already an element in the map, it either is something that was proved sat, which we prefer versus unproved, or another unproved, so we avoid redundant warnings.

Reply:
Expanding the auto will prevent using `[a, b]` and it's gonna look a lot uglier =/

Reply:
Is there an existing test that changed because of this?

Reply:
This also guarantees that the order of the warnings follows the AST ids, so many tests are changed because of that actually. The duplication part I'm not sure, will check.